ConsoleApp12/ConsoleApp12/Program.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ConsoleApp12
+{
+    internal class Program
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Введите число m от 5 до 20");
+            int m;
+
+            if (!TryInputNumber(out m))
+            {
+                Console.ReadKey();
+                return;
+            }
+
+            Console.WriteLine("Введите число n от 5 до 20");
+            int n;
+
+            if (!TryInputNumber(out n))
+            {
+                Console.ReadKey();
+                return;
+            }
+
+            if (m < 5 || m > 20 || n < 5 || n > 20)
+            {
+                Console.WriteLine("Числа не удовлетворяют неравенству 5 <= n, n <= 20");
+                Console.ReadKey();
+                return;
+            }
+
+            var matrix = new int[m, n];
+            var rnd = new Random();
+
+            for (int i = 0; i < matrix.GetLength(0); i++)
+                for (int j = 0; j < matrix.GetLength(1); j++)
+                    matrix[i, j] = rnd.Next(100);
+
+            Console.WriteLine();
+
+            PrintMatrix(matrix);
+
+            Console.WriteLine();
+
+            var hasZero = HasZeroElement(matrix);
+            if (hasZero.Item1)
+                Console.WriteLine($"0 найден в строке {hasZero.Item2} столбце {hasZero.Item3}");
+            else
+                Console.WriteLine("0 не найден");
+
+            Console.WriteLine();
+
+            var diffs = GetSum(matrix);
+
+            foreach (var diff in diffs)
+                Console.WriteLine($"строка {diff.Item1}: разность сумм = {diff.Item2}");
+
+            Console.ReadKey();
+        }
+
+        static bool TryInputNumber(out int number)
+        {
+            number = 0;
+            if (!int.TryParse(Console.ReadLine(), out int n))
+            {
+                Console.WriteLine("Ошибка ввода");
+                return false;
+            }
+
+            number = n;
+            return true;
+        }
+        static void PrintMatrix(int[,] matrix)
+        {
+            for (int i = 0; i < matrix.GetLength(0); i++)
+            {
+                for (int j = 0; j < matrix.GetLength(1); j++)
+                {
+                    Console.Write($"{matrix[i, j],2}");
+                }
+
+                Console.WriteLine();
+            }
```

Comment:
Элементы массива не отделяются пробелами при печати. Вот листинг:
```
Введите число m от 5 до 20
5
Введите число n от 5 до 20
5

5865362951
2091286392
74 2481972
 479317143
6594307358
```
```suggestion
        static void PrintMatrix(int[,] matrix)
        {
            for (int i = 0; i < matrix.GetLength(0); i++)
            {
                for (int j = 0; j < matrix.GetLength(1); j++)
                {
                    Console.Write($"{matrix[i, j],2} ");
                }

                Console.WriteLine();
            }
```